src/test/ui/feature-gates/feature-gate-const_transmute.stderr

```diff
@@ -0,0 +1,18 @@
+error[E0658]: The use of std::mem::transmute() is gated in constants (see issue #53605)
+  --> $DIR/feature-gate-const_transmute.rs:16:38
+   |
+LL | const TRANSMUTED_U32: u32 = unsafe { mem::transmute(Foo(3)) };
+   |                                      ^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: add #![feature(const_transmute)] to the crate attributes to enable
+
+error[E0015]: calls in constants are limited to constant functions, tuple structs and tuple variants
```

Comment:
hmm... this error duplication isn't too great. Maybe unconditionally do `is_const_fn = Some(def_id)`, so in case we emit the feature gate error, the rest of the code happily treats `transmute` as const fn (which doesn't matter, because we already emitted an error)